### PR TITLE
[#13] Integrate Travis CI

### PR DESCRIPTION
**Pull request by ayushjn20**
Add `.travis.yml` with 
* Golang support for `1.11.x` and `master` version.
* lint validation using `travis lint`. [[ref](https://github.com/travis-ci/travis.rb#lint)]

### CODE REVIEW

**Comment by ayushjn20** [comment]
@agentmilindu @rehrumesh @rehrumesh 
Please review.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+language: go
+
+go:
+  - "1.11.x"
+  - master
```
Comment (rehrumesh) [inline .travis.yml]: develop?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+env:
+  - DEP_VERSION=0.5.0
+  - NODE_TEST_IMAGE_VERSION=10.15.0
```
Comment (rehrumesh) [inline .travis.yml]: Do we need this env var? and DEV_VERSION?

File: .travis.yml (hunk 1 above)
Comment (ayushjn20) [reply .travis.yml]: Yes, for now. 1.12.x is coming with major changes in Feb 2019. Don't know for sure what are those.

File: .travis.yml (hunk 2 above)
Comment (ayushjn20) [reply .travis.yml]: Those are versions which may change with the course of time as `dep` hasn't released even the first version, and the other is just a test image which will be excluded from travis file once the issue #11 has been solved. I prefer to keep the versions as environment variables.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+services:
+  - docker
+
+before_install:
+  - curl -L -s https://github.com/golang/dep/releases/download/v${DEP_VERSION}/dep-linux-amd64 -o $GOPATH/bin/dep
+  - chmod +x $GOPATH/bin/dep
+
+  - docker pull node:${NODE_TEST_IMAGE_VERSION}
```
Comment (agentmilindu) [inline .travis.yml]: Why do we have to pull the Docker images? Dunner should pull the image itself, right?

Comment (agentmilindu) [reply .travis.yml]: Oh, got it!

**Pull request merged**